src/pt/mangastop/src/eu/kanade/tachiyomi/extension/pt/mangastop/MangaStop.kt

```diff
@@ -18,4 +22,17 @@ class MangaStop : MangaThemesia(
     override val client = super.client.newBuilder()
         .rateLimit(3)
         .build()
+
+    override fun pageListParse(document: Document): List<Page> {
+        val pages = super.pageListParse(document)
+            .filterNot { it.imageUrl?.contains("mihon", true) == true }
+
+        if (pages.isNotEmpty()) return pages
+
+        return MangaThemesia.JSON_IMAGE_LIST_REGEX.find(document.toString())
+            ?.groupValues?.get(1)
+            ?.let { json.parseToJsonElement(it).jsonArray }
+            ?.mapIndexed { i, el -> Page(i, document.location(), el.jsonPrimitive.content) }
```

Comment:
Do you need the page location? I didn't check if MangaThemesia does anything with it.

If it's not needed I would remove it.
```suggestion
            ?.mapIndexed { i, el -> Page(i, imageUrl = el.jsonPrimitive.content) }
```

Reply:
https://github.com/keiyoushi/extensions-source/blob/8769908c8413e22e6d2b3f0eea664add7d874935/lib-multisrc/mangathemesia/src/eu/kanade/tachiyomi/multisrc/mangathemesia/MangaThemesia.kt#L371-L377

 MangaThemesia parent class and imageRequest explicitly uses ```page.url``` to set the ```Referer``` header